### ISSUE TITLE
False positive for type statement with Literal

### ISSUE THREAD

**Issue by RomainBrault** [pylint-dev/pylint#9885]
### Bug description

```python
The following 

Let `test.py` be


"""False positive E1101"""

from typing import Literal


type Abc = Literal["a"]


print(Abc.__value__)


The above script work under python 3.12, __value__ is a valid member.

See:
 * https://docs.python.org/3/library/typing.html#typing.TypeAliasType.__value__
 * https://github.com/python/cpython/issues/112472
```

### Configuration

```ini
Default configuration
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:9:6: E1101: Class 'Literal' has no '__value__' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.12.4 (main, Jul 25 2024, 22:42:01) [Clang 18.1.8 ]
```

### OS / Environment

Ubuntu 24.04 LTS
